src/improvements/tasks/MultisigTask.sol

```diff
@@ -595,133 +605,138 @@ abstract contract MultisigTask is Test, Script, StateOverrideManager {
         require(
             childMultisig != address(0), "MultisigTask: Child multisig cannot be zero address when printing child hash."
         );
-
-        bytes memory callData = generateApproveMulticallData(actions);
         console.log("Child multisig: %s", getAddressLabel(childMultisig));
-
-        bytes memory encodedTxData = getEncodedTransactionData(childMultisig, callData);
-        require(encodedTxData.length == 66, "MultisigTask: encodedTxData length is not 66 bytes.");
-
-        bytes32 safeTxHash;
-        assembly {
-            // 66 bytes = (bytes1(0x19), bytes1(0x01), bytes32(domainSeparator()), bytes32(safeTxHash))
-            // Retrieve the last 32 bytes of encodedTxData (safeTxHash).
-            // Memory layout of encodedTxData:
-            // - The first 32 bytes store the length (66 bytes in this case).
-            // - The actual data starts at encodedTxData + 32.
-            // - The last 32 bytes of the data (safeTxHash) start at:
-            //   encodedTxData + 32 + (66 - 32) = encodedTxData + 66.
-            safeTxHash := mload(add(encodedTxData, mload(encodedTxData)))
-        }
-
-        bytes32 domainSeparator = GnosisSafeHashes.calculateDomainSeparator(block.chainid, childMultisig);
         console.log("Domain Hash:    ", vm.toString(domainSeparator));
-        console.log("Message Hash:   ", vm.toString(safeTxHash));
+        console.log("Message Hash:   ", vm.toString(messageHash));
     }
 
     /// @notice print the tenderly simulation payload with the state overrides
-    function printTenderlySimulationData(Action[] memory actions) internal view {
+    function printTenderlySimulationData(Action[] memory actions, address optionalChildMultisig) internal view {
+        // TODO: Support child nonce as a state override. Right now we always get the latest nonce.
+        // Use the max uint256 to indicate that the child multisig nonce is not provided (zero is a valid nonce).
+        uint256 childMultisigNonce =
+            optionalChildMultisig != address(0) ? _getNonce(optionalChildMultisig) : type(uint256).max;
         Simulation.StateOverride[] memory allStateOverrides =
-            getStateOverrides(parentMultisig, _getNonce(parentMultisig));
+            getStateOverrides(parentMultisig, _getNonce(parentMultisig), optionalChildMultisig, childMultisigNonce);
 
-        bytes memory txData = _execTransationCalldata(
-            parentMultisig, getMulticall3Calldata(actions), Signatures.genPrevalidatedSignature(msg.sender)
-        );
+        if (optionalChildMultisig != address(0)) {
+            bytes memory finalExec = getNestedSimulationMulticall3Calldata(actions, optionalChildMultisig);
+
+            console.log("\nSimulation link:");
+            Simulation.logSimulationLink({
+                _to: MULTICALL3_ADDRESS,
+                _data: finalExec,
+                _from: msg.sender,
+                _overrides: allStateOverrides
+            });
```

Comment:
Can I get a call to log out the payload as well, please?

```
            // Log the Tenderly JSON payload
            console.log("\nSimulation payload:");
            logTenderlySimulationPayload(finalExec, allStateOverrides);
```

Reply:
While I was going through this PR with Matt and Maurelian we decided to remove from the log output. Logging this in the terminal is unnecessary and creates clutter for the signers. When this feature is used in a later PR, instead of logging to the terminal, we should write it to a temporary file and then read it.

Reply:
@alcueca alternatively we could log this only if some env var like DEBUG_TENDERLY is set to true. 

Regardless, it's very important to minimize and optimize the information we present to signers, so I agree with Blaine here.

Reply:
Ok, all good, I'll find some way to get the payload out that doesn't clutter the output